rust/kernel/types.rs

```diff
@@ -91,3 +91,33 @@ impl<T: PointerWrapper + Deref> PointerWrapper for Pin<T> {
         Pin::new_unchecked(T::from_pointer(p))
     }
 }
+
+/// Runs a cleanup function/closure when dropped.
+///
+/// The [`Cleaner::dismiss`] function prevents the cleanup function from running.
+pub struct Cleaner<T: FnOnce()> {
```

Comment:
I think this is usually called `ScopeGuard`

Reply:
Indeed. I searched for the name but couldn't find it. Clearly I didn't try hard enough :)

Anyway, renamed to `ScopeGuard`.

Reply:
Agreed -- "scope guard" is the most common term in Rust, C++, etc. In C some may be accustomed to `cleanup` though (for the GCC attribute), and *maybe* we will get a `defer`/`guard` mechanism in standard C (the C committee is toying with that), but I think for us `ScopeGuard` is the most clear.